tuna/connection.py

```diff
@@ -174,14 +257,18 @@ def exec_command_unparsed(self, cmd, timeout=SSH_TIMEOUT, abort=None):
       if abort is not None and chk_abort_file(self.id, self.logger):
         self.logger.warning('Machine %s aborted command execution: %s', self.id,
                             cmd)
-        return None, None, None
+        return i_var, o_var, e_var
 
     self.logger.error('cmd_exec retries exhausted, giving up')
-    return None, None, None
+    return i_var, o_var, e_var
 
-  def exec_command(self, cmd, timeout=SSH_TIMEOUT, abort=None, proc_line=None):
+  def exec_command(self, cmd: str, timeout: int=int(SSH_TIMEOUT), abort: Optional[bool]=None,\
+  proc_line: Union[Any, None]= None) -> Tuple[ Union[int, Any], Union[ChannelFile, StringIO], \
```

Comment:
By removing the return at line 275, we can assure the return for this function will be (int, StringIO, StringIO).